packages/squiggle-lang/src/rescript/FR/FR_Danger_Aggregate.res

```diff
@@ -0,0 +1,171 @@
+open FunctionRegistry_Core
+open FunctionRegistry_Helpers
+
+let nameSpace = "Danger_Aggregate"
+let requiresNamespace = true
+
+module AggregateFs = {
+  module Helpers = {
+    let checker = (fn, xs, minLength) => {
+      switch E.A.length(xs) < minLength {
+      | true =>
+        Error(
+          "Aggregation method does not make sense with fewer than" ++
+          Belt.Int.toString(minLength) ++ " elements",
+        )
+      | false => {
+          let checkedIndividualXs = E.A.fmap(x =>
+            switch (x, x > 1.0, x < 0.0) {
+            | (0.0, _, _) => Error("0 is not a probability")
+            | (1.0, _, _) => Error("1 is not a probability")
+            | (_, true, _) => Error("Probabilities can't be higher than 1")
+            | (_, _, true) => Error("Probabilities can't be lower than 0")
+            | (_, false, false) => Ok(x)
+            }
+          , xs)
+          let checkedCollectiveXs = E.A.R.firstErrorOrOpen(checkedIndividualXs)
+          let result = switch checkedCollectiveXs {
+          | Ok(xs) => fn(xs)
+          | Error(e) => Error(e)
+          }
+          result
+        }
+      }
+    }
+    let probabilityToOdds = p => p /. (1.0 -. p)
+    let oddsToProbability = o => o /. (1.0 +. o)
+    let sum = xs => E.A.reduce(xs, 0.0, (a, b) => a +. b)
+    let mean = xs => sum(xs) /. Belt.Float.fromInt(E.A.length(xs))
+
+    let geomMean = (xs: array<float>) => {
+      let xsLogs = E.A.fmap(x => Js.Math.log2(x), xs)
+      let meanXsLogs = mean(xsLogs)
+      let answer = Js.Math.pow_float(~base=2.0, ~exp=meanXsLogs)
+      answer
+    }
+    let arithmeticMean = (xs: array<float>) => {
+      let sumXs = E.A.reduce(xs, 0.0, (a, b) => a +. b)
+      let meanXs = sumXs /. Belt.Float.fromInt(E.A.length(xs))
+      meanXs
+    }
```

Comment:
I'm fairly sure we already have code to do a lot of lines 37-50 (and if we don't, libraries we use do). Check in E.